DataFormats/Detectors/TRD/src/RawData.cxx

```diff
@@ -440,14 +444,16 @@ void setNumberOfTrackletsInHeader(o2::trd::TrackletMCMHeader& header, int number
 int nextmcmadc(unsigned int& bp, int channel)
 {
   //given a bitpattern (adcmask) find next channel with in the mask starting from the current channel;
-  while ((bp & (1 << channel)) == 0) {
-    channel++;
-    if (channel == 21) {
-      break;
-    }
+  if (bp == 0)
+    return 22;
+  int position = channel;
+  int m = 1 << channel;
+  while (!(bp & m)) {
```

Comment:
the newer one works properly, verified. unit tests to come.
the bit is fliped so that each time is simply looks for the next right most bit with a 1 in it, so at the end one has no bits and has a condition for finishing.

Reply:
To be clearer line 447,448 handles the case of no bits valid, it is possible to start on the left of the left most bit.
adding a termination at position 32;